collector/receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -573,3 +732,35 @@ func (r *Receiver) processRecords(ctx context.Context, method string, arrowConsu
 		return ErrUnrecognizedPayload
 	}
 }
+
+func (r *Receiver) acquireAdditionalBytes(ctx context.Context, uncompSize int64, response batchResp, sizeHeaderFound bool) error {
+	diff := uncompSize - response.bytesToRelease
+
+	var err error
+	if diff != 0 {
+		if sizeHeaderFound {
+			// a mismatch between header set by exporter and the uncompSize just calculated.
+			r.telemetry.Logger.Debug("mismatch between uncompressed size in receiver and otlp-pdata-size header", zap.Int("uncompsize", int(uncompSize)), zap.Int("otlp-pdata-size", int(response.bytesToRelease)))
+		} else if diff < 0 {
+			// proto.Size() on compressed request was greater than pdata uncompressed size.
+			r.telemetry.Logger.Debug("uncompressed size is less than compressed size", zap.Int("uncompressed", int(uncompSize)), zap.Int("compressed", int(response.bytesToRelease)))
+		}
```

Comment:
In my opinion, we must ensure that these structured logs contain enough information to enable us to ban buggy or misbehaving clients/exporters.

Reply:
If any significant work is needed for this, I'd like to file an issue and come back to it. OTLP doesn't have an in-band way to identify the sender -- I believe we have minimal information in the gRPC peer struct, plus any headers we add ourselves. If we could standardize on a useful header for the exporter to emit, then we could log it here.

Reply:
I created this issue https://github.com/open-telemetry/otel-arrow/issues/185

Reply:
However after thinking about this more I'm wondering how reliable the address is in the `client.Info` that we get from the incoming stream context. Does this address show up consistently https://github.com/open-telemetry/opentelemetry-collector/blob/main/client/client.go#L93

Reply:
Could this be enough information to identify harmful clients and block them from sending requests to the receiver?

Reply:
@lquerel @jmacd added the `client.Info.Addr` to the log message if it is available. This issue https://github.com/open-telemetry/otel-arrow/issues/185 can be used to track a followup to ban clients that are misusing the header, but I would like to know your thoughts on the best way to go about this. Thanks

Reply:
I don't know if client.Info is reliable. But we can probably find a way to get the client IP address at the gRPC connection level.